src/migration.ts

```diff
@@ -92,33 +92,36 @@ export class Migration {
    * @memberof Migration
    */
   public async config(opts?: IMigrationOptions): Promise<void> {
-
     this.options = Object.assign({}, this.options, opts);
 
     if (!this.options.logger && this.options.log) {
       // tslint:disable-next-line: no-console
       this.options.logger = (level: string, ...args) => console.log(level, ...args);
     }
+
     if (this.options.log === false) {
       // tslint:disable-next-line:no-empty
       this.options.logger = (level: string, ...args) => { };
     }
-    if (!(this.db instanceof Db) && !this.options.db) {
-      throw new ReferenceError('Option.db canno\'t be null');
+
+    let db: IDbProperties | Db = this.options.db || this.db;
+    if (!db) {
+      throw new ReferenceError('db option must be defined');
     }
-    let db: IDbProperties | Db;
-    if (this.options.db instanceof Db) {
-      db = this.options.db;
-    } else {
-      const options = { ...this.options.db.options };
+
+    // Check if connectionUrl exists. If it does, assume its IDbProperties object
```

Comment:
Per discussion with @stieg, the right solution is to make MongoDB a devDependency making it so the logic uses the consumer's own MongoDB package.